domain/ipaddress/ipaddress.go

```diff
@@ -0,0 +1,106 @@
+// Copyright 2024 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package ipaddress
+
+import corenetwork "github.com/juju/juju/core/network"
+
+// ConfigType represents the means via which
+// an IP address is configured, as recorded in the
+// ip_address_config_type lookup table.
+type ConfigType int
+
+const (
+	ConfigTypeUnknown ConfigType = iota
+	ConfigTypeDHCP
+	ConfigTypeDHCPv6
+	ConfigTypeSLAAC
+	ConfigTypeStatic
+	ConfigTypeManual
+	ConfigTypeLoopback
+)
+
+// MarshallConfigType converts an IP address config type to a db config type id.
+func MarshallConfigType(configType corenetwork.AddressConfigType) ConfigType {
+	switch configType {
+	case corenetwork.ConfigDHCP:
+		return ConfigTypeDHCP
+	case corenetwork.ConfigStatic:
+		return ConfigTypeStatic
+	case corenetwork.ConfigManual:
+		return ConfigTypeManual
+	case corenetwork.ConfigLoopback:
+		return ConfigTypeLoopback
+	}
+	return ConfigTypeUnknown
+}
+
+// Scope represents the scope of
+// an IP address, as recorded in
+// the ip_address_scope lookup table.
+type Scope int
```

Comment:
Ahh you must be using your VT05 in split screen.
```suggestion
// Scope represents the scope of an IP address, as recorded in the ip_address_scope lookup table.
type Scope int
```

or even
```suggestion
// Scope represents the scope of an IP address, as recorded in the
// ip_address_scope lookup table.
type Scope int
```

Reply:
ok